Update NEWS for 1.11 (#53681)
JuliaLang/julia: NEWS.md

  execution ([#52123]).
* Specifying a path in `JULIA_DEPOT_PATH` now results in the expansion of empty strings to
  omit the default user depot ([#51448]).
* Precompilation cache files are now relocatable and their validity is now verified through

>   a content hash of the `include` files instead of their `mtime` ([#49866]).

Reviewer: ```suggestion
  a content hash of their source files instead of their `mtime` ([#49866]).
```

* `copyuntil(out, io, delim)` and `copyline(out, io)` copy data into an `out::IO` stream ([#48273]).
* `eachrsplit(string, pattern)` iterates split substrings right to left ([#51646]).
* `Sys.username()` can be used to return the current user's username ([#51897]).

> * `Sys.isreadable(), Sys.iswriteable` can be used to check if the current user has access permissions

Reviewer 2: ```suggestion
* `Sys.isreadable(), Sys.iswriteable()` can be used to check if the current user has access permissions
```

> * `Sys.isreadable(), Sys.iswriteable()` can be used to check if the current user has access permissions

Author: ```suggestion
* `Sys.isreadable(), Sys.iswritable()` can be used to check if the current user has access permissions
```

Fixing spelling in accordance with #53699

* `filter` can now act on a `NamedTuple` ([#50795]).
* `Iterators.cycle(iter, n)` runs over `iter` a fixed number of times, instead of forever ([#47354]).
* `zero(::AbstractArray)` now applies recursively, so `zero([[1,2],[3,4,5]])` now produces the additive identity `[[0,0],[0,0,0]]` rather than erroring ([#38064]).

> * `include_dependency(path; track_content=true)` allows to switch from using `mtime`

Reviewer 3: ```suggestion
* `include_dependency(path; track_content=true)` allows switching from using `mtime`
```
